src/api.rs

```diff
@@ -135,6 +136,13 @@ pub fn prefix_is_subset(
     }
 }
 
+pub fn apps_is_subset(maybe_subset_apps: &Option<Vec<String>>, claimed_apps: &[String]) -> bool {
```

Comment:
My gut feeling is that it'd be slightly nicer if this were:
```suggestion
pub fn apps_is_subset(maybe_subset_apps: Option<&[String]>, claimed_apps: &[String]) -> bool {
```